### DISCUSSION TITLE
About parameter repeat in struct uv_timer_s

### DISCUSSION THREAD

**Discussion by muminglan** [libuv/libuv#3700] [Ideas]
code as below:
int uv_timer_again(uv_timer_t* handle) {
  if (handle->timer_cb == NULL)
    return UV_EINVAL;

  if (handle->repeat) {
    uv_timer_stop(handle);
    uv_timer_start(handle, handle->timer_cb, handle->repeat, handle->repeat);
  }
  return 0;
}
My question is repeat just like a switch to enable repeat or not, Why not use bool? Because timeout in struct uv_timer_s can do this。is repeat redundant?

**Comment by bnoordhuis** [comment]
> Why not use bool?

It could (have - not anymore) but a configurable repeat is a little more flexible.

Honestly, if I'd redesign libuv from scratch today, I'd probably leave out repeat altogether.
